### DISCUSSION TITLE
Changing Default Marlin Tool Number

### DISCUSSION THREAD

**Discussion by O4k** [supermerill/SuperSlicer#1828] [Q&A] [answered]
I have a Bigrep Studio with Marlin Firmware and a dual extruder. I am trying to change the default tool number for the gcode and it appears that the default tool number starts at 'T1'.  This is unfortunate since the BigRep's default extruder is 'T0'.  Short of creating a post-script specifically to change this number, I have not found any easy way to adjust the tool number.

Is there any way to change the tool number in superslicer when running Marlin?

**Comment by supermerill** [comment, marked as answer]
the `TX` command?
put `T{next_extruder+1}` in the custom toolchange gcode.

**Reply by O4k** [reply]
Thank you! I used this with `T{next_extruder-1}` and it seemed to work well.

I forget all the ways you can use these variables!

**Reply by supermerill** [reply]
weird, I though that the extruder id began at 0 ...

**Reply by O4k** [reply]
I have been messing with quite a few settings on the printer and in the slicer so take all of this with a grain of salt. 
I've gotten inconsistent results; about half of the time `T1` was mentioned at the at the beginning of the gcode file, and the other times it is not even listed.

To really iron this out, I will have to be more thorough when testing it out. For now, this workaround will work.

**Reply by O4k** [reply]
Figured it out! 

This ended up being a problem with the firmware and my assumptions about the GCode not Superslicer. It appears that the labeling of objects at the start of the print is unsupported by this printer's version of marlin and so the "T1" part label was being confused as a command to switch extruders.
